src/lib/Libtpp/tpp_client.c

```diff
@@ -673,7 +673,10 @@ tpp_init(struct tpp_config *cnf)
 		return -1;
 	}
 
-	snprintf(log_buffer, TPP_LOGBUF_SZ, "TPP leaf node names = %s", tpp_conf->node_name);
+	if (tpp_conf->node_name)
```

Comment:
Instead of only printing when node_name is not null, please do the check before and return from the function. So something like this:

f (tpp_conf->node_name == NULL) {
   snprintf(log_buffer, TPP_LOGBUF_SZ, "TPP leaf node name is NULL");
   tpp_log_func(LOG_CRIT, NULL, log_buffer);
   return -1;
}

snprintf(log_buffer, TPP_LOGBUF_SZ, "TPP leaf node names = %s", tpp_conf->node_name);
tpp_log_func(LOG_CRIT, NULL, log_buffer);

Reply:
Thanks for your comment @subhasisb 
I have addressed as suggested. please have a look.